### PR TITLE
[Device Registry] Hotfixes AND metadata(DB) updates + migrations

### PR DESCRIPTION
**Pull request by Baalmart**
**_WHAT DOES THIS PR DO?_**

- [x] Resolve a runtime error for Kafka consumption. 
- [x] Update the schema for Sites to address type issue for one field (`land_use`)
- [x] Add appropriate input validations for the NETWORK field during update processes for Sites and Devices
- [x] Update the caching strategy/ID for when one queries using `lat_long`.
- [x] Delete unused configuration files for Kafka
- [x] Create scripts to be used for BULK addition and update of new data into device registry.
- [x] Do data entry and update the `device_codes` and `site_codes` for ALL devices under the AirQo tenant. 
- [x] Migrate KCCA sites, devices and AirQlouds to AirQo tenant.
- [x] Update the Sites and Devices to have the data for the `site_codes` and `device_codes` accordingly.
- [x] With respect to international expansion, modify the required fields during device creation. Focus on the SOFT creation endpoint for devices.
- [x] To reflect international focus, make appropriate schema changes for Devices by mostly reducing the number of required fields. Ensure that this change is also reflected in the DB indixes for staging.
- [x] Deletion of unused code and files

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [AN-227]

**_HOW DO I TEST OUT THIS PR?_**
```
cd src/device-registry
npm install
npm run stage-mac
```

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

For all the GET endpoints, use the tenant and network query parameters accordingly. Eg: `?tenant=airqo&network=kcca`

- [x] [update site](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-f979140c-0048-4834-9e24-ba237bf2ed38)
- [x] [get events (v1)](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-2f1360d4-2eaa-40b7-9ad4-eef1b03abbce) using `lat_long` query parameter.
- [x] [SOFT update Device](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-45e2d8f1-b0a7-476c-950c-e7e9576bbda7)
- [x] [Get Sites](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-d99d2b14-eab1-448a-8bc5-369c76a9579e)
- [x] [Get Devices](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-ec2c7c03-6429-43b6-b000-c7469553748d)
- [x] [Get AirQlouds](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-9a62c9c5-f1b2-4206-8f45-a8db3bbdce0d)
- [x] [Get Activities](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-006535db-c82a-404c-81ae-9cd71667965c)

**_SPECIAL CONSIDERATIONS AND PENDING ISSUES?:_** 

1. The NETWORK field will NOT be mandatory during Device and Site Creation. It is just added to those entities afterwards. Because a Site or Device can belong to 0 (Zero) or more networks.
2. Initially, all the data migration will be done from a staging ENV and these will be migrated to production when everything meets expectations after thorough QA tests are carried out.
3. In a future PR, we shall ensure that the QR code for device-registry returns all the Device information.
4. We shall also ensure that the measurements from Kafka are actually stored in the microservice's DB
5. We shall also add the 3 sites for AirNow which were not valid for our system and "deploy" them accordingly.

[AN-227]: https://airqoteam.atlassian.net/browse/AN-227?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/device-registry/controllers/create-device.js [hunk 1]
```diff
@@ -377,83 +376,6 @@ const device = {
     }
   },
 
-  updateAccessCode: async (req, res) => {
-    try {
```
Comment (Baalmart) [inline src/device-registry/controllers/create-device.js:381]: delete unused functionality

File: src/device-registry/models/Site.js [hunk 2]
```diff
@@ -136,10 +137,12 @@ const siteSchema = new Schema(
       type: String,
       trim: true,
     },
-    land_use: {
-      type: String,
-      trim: true,
-    },
+    land_use: [
```
Comment (Baalmart) [inline src/device-registry/models/Site.js:139]: addressing the `land_use` type issue, aligning it with what is expected. cc: @NoahNsimbe

File: src/device-registry/utils/create-event.js [hunk 3]
```diff
@@ -1105,7 +1107,7 @@ const createEvent = {
       external ? external : "noExternal"
     }_${airqloud ? airqloud : "noAirQloud"}_${
       airqloud_id ? airqloud_id : "noAirQloudID"
-    }`;
+    }_${lat_long ? lat_long : "noLatLong"}`;
```
Comment (Baalmart) [inline src/device-registry/utils/create-event.js:1110]: Updating the caching ID generation to cater for the case of querying measurements by `lat_long`

File: src/device-registry/controllers/create-activity.js [hunk 4]
```diff
@@ -222,6 +224,138 @@ const activity = {
       });
     }
   },
+
+  bulkAdd: async (req, res) => {
+    try {
```
Comment (Baalmart) [inline src/device-registry/controllers/create-activity.js:229]: controller logic for bulk addition of activities

File: src/device-registry/controllers/create-activity.js [hunk 5]
```diff
@@ -222,6 +224,138 @@ const activity = {
...
+      return res.status(HTTPStatus.INTERNAL_SERVER_ERROR).json({
+        success: false,
+        message: "Internal Server Error",
+        errors: { message: error.message },
+      });
+    }
+  },
+
+  bulkUpdate: async (req, res) => {
+    try {
```
Comment (Baalmart) [inline src/device-registry/controllers/create-activity.js:295]: controller logic for bulk update of acitivities

File: src/device-registry/.gitignore [hunk 6]
```diff
@@ -2,4 +2,8 @@ node_modules
 controllers/rsa_private.pem
 .env
 device-reg-secret.yaml
-log
\ No newline at end of file
+log
+scripts/data-airqlouds.js
```
Comment (Baalmart) [inline src/device-registry/.gitignore:6]: just ignoring the local files which contain the data for carrrying out bulk operations

File: src/device-registry/controllers/create-airqloud.js [hunk 7]
```diff
@@ -9,8 +9,49 @@ const logger = log4js.getLogger(
   `${constants.ENVIRONMENT} -- create-airqloud-controller`
 );
 const httpStatus = require("http-status");
+const bulkCreateUtil = require("../scripts/bulk-create");
 
 const createAirqloud = {
+  bulkCreate: async (req, res) => {
+    try {
```
Comment (Baalmart) [inline src/device-registry/controllers/create-airqloud.js:16]: controller logic for bulk addition of AirQlouds

File: src/device-registry/controllers/create-device.js [hunk 8]
```diff
@@ -11,8 +11,96 @@ const log4js = require("log4js");
...
   `${constants.ENVIRONMENT} -- create-device-controller`
 );
+const bulkUpdateUtil = require("../scripts/bulk-update");
+const bulkCreateUtil = require("../scripts/bulk-create");
+const httpStatus = require("http-status");
 
 const device = {
+  bulkCreate: async (req, res) => {
+    try {
+      const hasErrors = !validationResult(req).isEmpty();
```
Comment (Baalmart) [inline src/device-registry/controllers/create-device.js:21]: controller logic for bulk addition of Devices

File: src/device-registry/controllers/create-device.js [hunk 9]
```diff
@@ -11,8 +11,96 @@ const log4js = require("log4js");
...
+    } catch (error) {
+      logObject("error", error);
+      res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+        success: false,
+        message: "updating all the metadata",
+        errors: { message: error.message },
+      });
+    }
+  },
+  bulkUpdate: async (req, res) => {
```
Comment (Baalmart) [inline src/device-registry/controllers/create-device.js:62]: controller logic for bulk update of Devices

File: src/device-registry/controllers/create-site.js [hunk 10]
```diff
@@ -13,8 +13,122 @@ const httpStatus = require("http-status");
 const logger = log4js.getLogger(
   `${constants.ENVIRONMENT} -- create-site-controller`
 );
+const bulkUpdateUtil = require("../scripts/bulk-update");
+const bulkDeleteUtil = require("../scripts/bulk-delete");
+const bulkCreateUtil = require("../scripts/bulk-create");
 
 const manageSite = {
+  bulkCreate: async (req, res) => {
+    try {
```
Comment (Baalmart) [inline src/device-registry/controllers/create-site.js:22]: controller logic for bulk addition of Sites

File: src/device-registry/controllers/create-site.js [hunk 11]
```diff
@@ -13,8 +13,122 @@ const httpStatus = require("http-status");
...
+      }
+    } catch (error) {
+      return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+        message: "internal server error",
+      });
+    }
+  },
+  bulkUpdate: async (req, res) => {
+    try {
+      const hasErrors = !validationResult(req).isEmpty();
```
Comment (Baalmart) [inline src/device-registry/controllers/create-site.js:62]: controller logic for bulk update of Sites

File: src/device-registry/controllers/create-site.js [hunk 12]
```diff
@@ -13,8 +13,122 @@ const httpStatus = require("http-status");
...
+      }
+    } catch (error) {
+      return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+        message: "internal server error",
+      });
+    }
+  },
+
+  bulkDelete: async (req, res) => {
+    try {
```
Comment (Baalmart) [inline src/device-registry/controllers/create-site.js:99]: controller logic for bulk deletion of Sites

File: src/device-registry/models/Airqloud.js [hunk 13]
```diff
@@ -126,6 +126,12 @@ const airqloudSchema = new Schema(
       required: [true, "name is required!"],
       unique: true,
     },
+    airqloud_codes: [
```
Comment (Baalmart) [inline src/device-registry/models/Airqloud.js:129]: new field to the AirQloud schema/entity

File: src/device-registry/models/Airqloud.js [hunk 14]
```diff
@@ -150,7 +156,10 @@ const airqloudSchema = new Schema(
       required: [true, "isCustom is required!"],
     },
     metadata: { type: metadataSchema },
-    network: { type: String },
+    network: {
```
Comment (Baalmart) [inline src/device-registry/models/Airqloud.js:159]: new schema field for AirQloud entity

File: src/device-registry/models/Location.js [hunk 15]
```diff
@@ -74,7 +74,10 @@ const locationSchema = new Schema(
       default: false,
     },
     metadata: { type: metadataSchema },
-    network: { type: String },
+    network: {
```
Comment (Baalmart) [inline src/device-registry/models/Location.js:77]: new field for Location entity

File: src/device-registry/models/Site.js [hunk 16]
```diff
@@ -386,30 +388,30 @@ siteSchema.statics = {
...
+        delete data.site_tags;
+        delete data.nearest_tahmo_station;
         return {
           success: true,
           data,
           message: "site created",
           status: HTTPStatus.CREATED,
         };
-      } else {
+      } else if (isEmpty(createdSite)) {
```
Comment (Baalmart) [inline src/device-registry/models/Site.js:411]: resolving a runtime error

File: src/device-registry/models/SiteActivity.js [hunk 17]
```diff
@@ -10,8 +10,17 @@ const activitySchema = new Schema(
     site_id: { type: ObjectId },
     date: { type: Date },
     description: { type: String, trim: true },
-    network: { type: String, trim: true },
+    network: {
+      type: String,
+      trim: true,
+    },
     activityType: { type: String, trim: true },
+    activity_codes: [
```
Comment (Baalmart) [inline src/device-registry/models/SiteActivity.js:18]: We should now be able to query activities using these codes

File: src/device-registry/scripts/bulk-create.js [hunk 18]
```diff
@@ -0,0 +1,301 @@
+const getDevices = require("./data-devices");
+const getSites = require("./data-sites");
```
Comment (Baalmart) [inline src/device-registry/scripts/bulk-create.js:2]: The scripts for bulk creation

File: src/device-registry/scripts/bulk-delete.js [hunk 19]
```diff
@@ -0,0 +1,202 @@
+/**
+ * For the device_codes....
+ * Fetch the available and get their IDS, store them in an array.
+ * 
+ * Iterate over each of the IDS and
+ just delete all the device_codes accordingly.
+ * 
+ Repeat the same procedure for the site_codes...
+ *
```
Comment (Baalmart) [inline src/device-registry/scripts/bulk-delete.js:9]: The scripts for bulk deletions

File: src/device-registry/scripts/bulk-update.js [hunk 20]
```diff
@@ -0,0 +1,241 @@
+/**
+ * For the device_codes....
+ * Fetch the available and get their IDS, store them in an array.
+ * 
+ * Iterate over each of the IDS and
+ just update all the device_codes accordingly.
```
Comment (Baalmart) [inline src/device-registry/scripts/bulk-update.js:6]: The scripts for bulk updates

File: src/device-registry/utils/create-airqloud.js [hunk 21]
```diff
@@ -125,8 +125,13 @@ const createAirqloud = {
       let requestForCalucaltionAirQloudCenter = {};
       requestForCalucaltionAirQloudCenter["body"] = {};
       requestForCalucaltionAirQloudCenter["query"] = {};
-      requestForCalucaltionAirQloudCenter["body"]["coordinates"] =
-        modifiedBody.location.coordinates[0];
+      if (
+        !isEmpty(modifiedBody.location) &&
```
Comment (Baalmart) [inline src/device-registry/utils/create-airqloud.js:129]: Handling an edge case when creating an AirQloud

File: src/device-registry/utils/generate-filter.js [hunk 22]
```diff
@@ -747,6 +747,7 @@ const generateFilter = {
       airqloud_id,
```
Comment (Baalmart) [inline src/device-registry/utils/generate-filter.js:747]: With the new query fields (`device_codes`, `activity_codes`, etc) in place, making the relevant updates to our filter util

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Pull request merged**